WordPress/Classes/ViewRelated/Stats/Shared Views/Stats Detail/SiteStatsDetailsViewModel.swift

```diff
@@ -0,0 +1,124 @@
+import Foundation
+import WordPressFlux
+
+/// The view model used by SiteStatsDetailTableViewController to show
+/// all data for a selected stat.
+///
+class SiteStatsDetailsViewModel: Observable {
+
+    // MARK: - Properties
+
+    let changeDispatcher = Dispatcher<Void>()
+
+    private typealias Style = WPStyleGuide.Stats
+
+    private var statSection: StatSection?
+    private weak var detailsDelegate: SiteStatsDetailsDelegate?
+
+    private let insightsStore = StoreContainer.shared.statsInsights
+    private var insightsReceipt: Receipt?
+    private var insightsChangeReceipt: Receipt?
+
+    init(detailsDelegate: SiteStatsDetailsDelegate) {
+        self.detailsDelegate = detailsDelegate
+    }
+
+    func fetchDataFor(statSection: StatSection) {
+        self.statSection = statSection
+
+        if StatSection.allInsights.contains(statSection) {
+            guard let storeQuery = queryForInsightStatSection(statSection) else {
+                return
+            }
+
+            insightsReceipt = insightsStore.query(storeQuery)
+            insightsChangeReceipt = insightsStore.onChange { [weak self] in
+                self?.emitChange()
+            }
+        } else {
+
```

Comment:
Do we anticipate some additional behavior here?

If so, perhaps we could annotate it as such in the same way you did for `SiteStatsDetailTableViewController`?
If not, perhaps, the `if` control flow could be replaced with `guard`?

Reply:
Dang, I really thought I put a TODO in there. Yes, this else will be used to handle Period queries in the near future. I'll add a note.